src/geometry/euclidean/angle/unoriented/affine.lean

```diff
@@ -439,4 +439,53 @@ lemma angle_lt_pi_of_not_collinear {p₁ p₂ p₃ : P} (h : ¬collinear ℝ ({p
   ∠ p₁ p₂ p₃ < π :=
 (angle_le_pi _ _ _).lt_of_ne $ angle_ne_pi_of_not_collinear h
 
+/-- The sine of the angle between three points is zero if and only if the angle is 0 or π. -/
+lemma sin_eq_zero_iff_angle_eq_zero_or_angle_eq_pi {p₁ p₂ p₃ : P} :
+  real.sin (∠ p₁ p₂ p₃) = 0 ↔ ∠ p₁ p₂ p₃ = 0 ∨ ∠ p₁ p₂ p₃ = π :=
```

Comment:
This should be stated in `geometry.euclidean.angle.unoriented.basic` as a lemma for angles between vectors. Then the version for angles between points can be deduced immediately from that (being defeq to a particular case of the lemma for angles between vectors).